src/legacy/ui/public/agg_types/agg_params.ts

```diff
@@ -17,46 +17,43 @@
  * under the License.
  */
 
-import { IndexedArray } from '../indexed_array';
+import { AggParamType } from 'ui/agg_types/param_types/agg';
 import { FieldParamType } from './param_types/field';
 import { OptionedParamType } from './param_types/optioned';
 import { StringParamType } from './param_types/string';
 import { JsonParamType } from './param_types/json';
 import { BaseParamType } from './param_types/base';
-import { createLegacyClass } from '../utils/legacy_class';
+import { AggConfig } from '../vis/agg_config';
+import { AggConfigs } from '../vis/agg_configs';
+
+export type AggParam = BaseParamType;
+
+export interface AggParamOption {
+  val: string;
+  display: string;
+  enabled?(agg: AggConfig): void;
```

Comment:
I realize you just copied this over, but I don't understand this particular line... So `enabled` is an optional param, which is just a function that takes in an `AggConfig` and doesn't return? 

Based on the name `enabled`, I would've expected something more like this:
```ts
enabled?: (agg: AggConfig) => boolean;
```

Reply:
yeah i just copied it over but i agree this should return boolean